controllers/subscription_controller.go

```diff
@@ -310,6 +312,18 @@ func (r *SubscriptionReconciler) setOperatorCondition(ctx context.Context, logge
 		}
 	}
 
+	// Make operator not upgradeable if ODF version is n+1 ahead of OCP version
```

Comment:
at multiple places I see `n`, I suppose `n` is referring to OCP, better be explicit, if OCP is at 4.Y and ODF is at 4.(Y+1) do not allow ODF to upgrade further.